tests/performance/optimized_select_final.xml

```diff
@@ -0,0 +1,20 @@
+<test>
+    <settings>
+        <do_not_merge_across_partitions_select_final>1</do_not_merge_across_partitions_select_final>
+    </settings>
+
+    <create_query>
+        CREATE TABLE optimized_select_final (t DateTime, x Int32)
+        ENGINE = ReplacingMergeTree()
+        PARTITION BY toYYYYMM(t) ORDER BY x
+    </create_query>
+
+    <fill_query>INSERT INTO optimized_select_final SELECT toDate('2000-01-01'), number FROM numbers(1000000)</fill_query>
+
+    <fill_query>INSERT INTO optimized_select_final SELECT toDate('2020-01-01'), number FROM numbers(1000000)</fill_query>
```

Comment:
It makes sense to add `optimize table` as the final fill query -- this way, you can be sure that all background merges have completed before the test queries start, and will get more stable results.